drivers/ina3221/ina3221.c

```diff
@@ -0,0 +1,928 @@
+
+#include "include/ina3221_defines.h"
+#include "include/ina3221_params.h"
+#include "include/ina3221_regs.h"
+#include "ina3221.h"
+
+struct channel_info
+{
+    uint8_t  chi_reg_shunt;
+    uint8_t  chi_reg_bus;
+    uint8_t  chi_reg_crit_alert_limit;
+    uint8_t  chi_reg_warn_alert_limit;
+};
+
+static struct channel_info _chi[INA3221_NUM_CH] = {
+    (struct channel_info){
+        .chi_reg_shunt            = INA3221_REG_CH1_SHUNT_VOLTAGE,
+        .chi_reg_bus              = INA3221_REG_CH1_BUS_VOLTAGE,
+        .chi_reg_crit_alert_limit = INA3221_REG_CH1_CRIT_ALERT_LIMIT,
+        .chi_reg_warn_alert_limit = INA3221_REG_CH1_WARN_ALERT_LIMIT
+    },
+    (struct channel_info){
+        .chi_reg_shunt            = INA3221_REG_CH2_SHUNT_VOLTAGE,
+        .chi_reg_bus              = INA3221_REG_CH2_BUS_VOLTAGE,
+        .chi_reg_crit_alert_limit = INA3221_REG_CH2_CRIT_ALERT_LIMIT,
+        .chi_reg_warn_alert_limit = INA3221_REG_CH2_WARN_ALERT_LIMIT
+    },
+    (struct channel_info){
+        .chi_reg_shunt            = INA3221_REG_CH3_SHUNT_VOLTAGE,
+        .chi_reg_bus              = INA3221_REG_CH3_BUS_VOLTAGE,
+        .chi_reg_crit_alert_limit = INA3221_REG_CH3_CRIT_ALERT_LIMIT,
+        .chi_reg_warn_alert_limit = INA3221_REG_CH3_WARN_ALERT_LIMIT
+    },
+};
+
+/**
+ * @brief Convert register value to shunt voltage in uV
+ * 
+ * @param[in]   reg_val Register value
+ * 
+ * @pre         reg_val must be in host byte order
+ * 
+ * @returm      Shunt voltage in uV
+ */
+static inline int32_t reg_val_to_shunt_voltage_uV(int16_t reg_val)
+{
+    //assert(reg_val >= INA3221_MIN_SHUNT_REG_VAL);
+    assert(reg_val <= INA3221_MAX_SHUNT_REG_VAL);
+    
+    int32_t shunt_voltage_uV;
+    if(reg_val < 0)
+    {
+        shunt_voltage_uV = (~(reg_val) + 1) / 8 * INA3221_SHUNT_VOLTAGE_PRECISION_uV;
+        shunt_voltage_uV *= -1;
+    }
+    else
+    {
+        shunt_voltage_uV =  reg_val / 8 * INA3221_SHUNT_VOLTAGE_PRECISION_uV;
+    }
+    return shunt_voltage_uV;
+}
+
+/**
+ * @brief Convert shunt voltage in uV to register value
+ * 
+ * @param[in]   s_uV Shunt voltage in uV
+ * 
+ * @post        Output register value is in host byte order
+ * 
+ * @return      Register value
+ */
+static inline int16_t shunt_voltage_uV_to_reg_val(int32_t s_uV)
+{
+    //assert(s_uV >= INA3221_MIN_SHUNT_uV);
+    assert(s_uV <= INA3221_MAX_SHUNT_uV);
+
+    int16_t reg_val;
+    if(s_uV < 0)
+    {
+        s_uV *= (-1);
+        reg_val = ((~(s_uV / INA3221_SHUNT_VOLTAGE_PRECISION_uV * 8)) + 1);
+    }
+    else
+    {
+        reg_val = s_uV / INA3221_SHUNT_VOLTAGE_PRECISION_uV * 8;
+    }
+    return reg_val;
+}
+
+/**
+ * @brief Convert register value to bus voltage in mV
+ * 
+ * @param[in]   reg_val Register value
+ * 
+ * @pre         reg_val must be in host byte order
+ * 
+ * @returm      Shunt voltage in mV
+ */
+static inline int16_t reg_val_to_bus_voltage_mV(int16_t reg_val)
+{
+    //assert(reg_val >= INA3221_MIN_BUS_REG_VAL);
```

Comment:
Why commented out?

Reply:
Compiler reported it will always evaluate true. I will remove it.